File: Sum/InorderTraversal.java
public class InorderTraversal{
	// Iterative Solution 
	public List<Integer> inorderTraversal(TreeNode root){
		List<Integer> result = new ArrayList<>();
		Stack<TreeNode> stack = new Stack<TreeNode>();
		TreeNode cur = root;

		while(cur!=null || !stack.empty()){
			while(cur!=null){
				stack.add(cur);
				cur = cur.left;
			}
			cur = stack.pop();
			list.add(cur.val);
			cur = cur.right;
		}
		return list;
	}

	
}

File: Sum/Tree/inordertraversal.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val) {
 *     this.val = val;
 *     this.left = this.right = null;
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number[]}
 */
 // iterative appraoch, reduce the while loop 
var inorderTraversal = function(root) {
    var stack = [], res = [], cur=root;
    while(stack.length>0 || cur!==null){
        // no need the while loop here 
        if(cur!==null){
            stack.push(cur);
            cur = cur.left;
        }else{
            var node = stack.pop();
            res.push(node.val);
            cur = node.right;
        }
    }
    return res;
};